src/pybind/mgr/mgr_module.py

```diff
@@ -286,57 +301,75 @@ def device_class_counts(self):
 class CLICommand(object):
     COMMANDS = {}  # type: Dict[str, CLICommand]
 
-    def __init__(self, prefix, args="", desc="", perm="rw"):
+    def __init__(self, prefix, perm="rw"):
         self.prefix = prefix
-        self.args = args
-        self.args_dict = {}
-        self.desc = desc
         self.perm = perm
         self.func = None  # type: Optional[Callable]
-        self._parse_args()
-
-    def _parse_args(self):
-        if not self.args:
-            return
-        args = self.args.split(" ")
-        for arg in args:
-            arg_desc = arg.strip().split(",")
-            arg_d = {}
-            for kv in arg_desc:
-                k, v = kv.split("=")
-                if k != "name":
-                    arg_d[k] = v
-                else:
-                    self.args_dict[v] = arg_d
+        self.arg_spec = {}    # type: Dict[str, Any]
+
+    KNOWN_ARGS = '_', 'self', 'mgr', 'inbuf'
+
+    @staticmethod
+    def load_func_metadata(f):
+        desc = inspect.getdoc(f) or ''
+        full_argspec = inspect.getfullargspec(f)
```

Comment:
What about inferring the command name? That way, `CLICommand` could probably be a decorator without args and infer everything from the function signature?

```suggestion
__module_prefix__ = 'dashboard'
...
    def load_func_metadata(f):
        if __module_prefix__:
            self.prefix = f'{__module_prefix__} {f.__name__.replace('_', '-')}'
        desc = inspect.getdoc(f) or ''
        full_argspec = inspect.getfullargspec(f)
```

Reply:
@epuertat indeed! that's exactly what i've been thinking. but i'd like to leave it for another PR.

Reply:
No problem... In fact this is less trivial than I initially though as we'd need access to the instantiated `MgrModule.module_name` property...